File: CTask/src/org/ShapeSHifter420/Fabric/TechPushFactory.java
package org.ShapeSHifter420.Fabric;

import org.ShapeSHifter420.Pushes.IPush;
import org.ShapeSHifter420.Pushes.TechPush;

import java.util.Map;

public class TechPushFactory implements IPushFactory{
    @Override
    public IPush create(Map<String, String> conf) {
        String text = conf.getOrDefault("text", "");
        float osversion = conf.containsKey("os_version")? Float.parseFloat(conf.get("os_version")) : -1;
        return new TechPush(text,osversion);
    }
}

File: CKotlin/Factory/TechPushFactory.kt
package Factory

import Pushes.IPush
import Pushes.TechPush

object TechPushFactory : IPushFactory {
    override fun made(param: Map<String, String>): IPush = TechPush(
        text = param["text"] ?: "none",
        osVersion = param["os_version"]?.toInt() ?: 0
    )
}